### PR TITLE
ENH plotting module

### PR DESCRIPTION
**Pull request by tomMoral**
Move the `viz` module to `plotting` and add some functionality to generate benchmark curves with `plotly`.

### CODE REVIEW

File: benchopt/cli.py [hunk 1]
```diff
@@ -147,28 +146,29 @@ def run(benchmark, solver_names, forced_solvers, dataset_names,
...
               help='Timeout a solver when run for more than <timeout> seconds')
-@click.option('--no-display',
-              is_flag=True,
-              help="If this flag is set, do not display the plot on the "
-              "screen.")
+@click.option('--display/--no-display', default=True,
+              help="Whether or not to display the plot on the screen.")
+@click.option('--plotly', is_flag=True,
+              help="If this flag is set, generate figure as HTML with plotly. "
+              "This option does not work with all kind of plots and requires "
```
Comment (tomMoral) [inline benchopt/cli.py:160]:
```suggestion
              "This option does not work with all plot kinds and requires "
```

File: benchopt/cli.py [hunk 2]
```diff
@@ -147,28 +146,29 @@ def run(benchmark, solver_names, forced_solvers, dataset_names,
...
+              "to have plotly installed.")
```
Comment (tomMoral) [inline benchopt/cli.py:161]:
```suggestion
              "to have installed `plotly`.")
```

File: benchopt/plotting/plot_objective_curve.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+    y_label = r"$F(x)$"
+    if suboptimality:
+        eps = 1e-10
+        y_label = r"$F(x) - F(x*)$"
+        c_star = df['objective_value'].min() - eps
+        df['objective_value'] -= c_star
+
+    if relative:
+        y_label = r"$\frac{F(x) - F(x*)}{F(x^0) - F(x*)}$"
```
Comment (tomMoral) [inline benchopt/plotting/plot_objective_curve.py:50]:
```suggestion
        obj = y_label[1:-1]
        y_label = r"$\frac{{{}}}{{{}}}$".format(
            obj, obj.replace('F(x)', 'F(x^0)')
        )
```

File: benchopt/plotting/plot_objective_curve.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
+import matplotlib.pyplot as plt
+
+from .helpers_compat import get_figure
+from .helpers_compat import add_h_line
+from .helpers_compat import fill_between_x
+
+
+CMAP = plt.get_cmap('tab20')
```
Comment (tomMoral) [inline benchopt/plotting/plot_objective_curve.py:8]:
```suggestion
# XXX: this cmap is only valid for benchmark that have less than 20 solvers
# Finding a solution to get a good color palette for any number (<50) solvers
# would be nice.
CMAP = plt.get_cmap('tab20')
```

**Review by tomMoral — COMMENTED** [review]
A few updates

**Review by agramfort — APPROVED** [review]
nice !

**Pull request merged**